### ISSUE TITLE
[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] provisioning should provision storage with mount options failing on Fedora/RHEL and CRI-O

### ISSUE THREAD

**Issue by runcom** [kubernetes/kubernetes#70239]
**Which jobs are failing**:

CRI-O job in our CI using kubernetes master tests

**Which test(s) are failing**:

[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] provisioning should provision storage with mount options

**Since when has it been failing**:

since this PR has been merged: https://github.com/kubernetes/kubernetes/pull/69441

**Testgrid link**:

link to our job is here:

https://openshift-gce-devel.appspot.com/build/origin-federated-results/pr-logs/pull/kubernetes-sigs_cri-o/1870/test_pull_request_crio_e2e_fedora/1368/
https://openshift-gce-devel.appspot.com/build/origin-federated-results/pr-logs/pull/kubernetes-sigs_cri-o/1874/test_pull_request_crio_e2e_fedora/1380/
https://openshift-gce-devel.appspot.com/build/origin-federated-results/pr-logs/pull/kubernetes-sigs_cri-o/1874/test_pull_request_crio_e2e_rhel/1557/

**Reason for failure**:

we don't know the reason of the failure but it doesn't seem related to CRI-O. It's constantly reproducible also.

**Anything else we need to know**:

/cc @jberkus @mkimuram @msau42

/kind failing-test

**Comment by runcom** [comment]
I'm still looking at it on our side

**Comment by mkimuram** [comment]
@runcom 

> [sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (default fs)] provisioning should provision storage with mount options

This test is newly added in this commit. It tests that nfs external dynamic provisioner can provision a volume with specific mount options and the volume is properly mounted in the pod. The mount options specified in this test are "vers=4.1", "proto=tcp", and "nosuid". (https://github.com/kubernetes/kubernetes/pull/69441/commits/4fcb36e205b0a589806debf9ea20d873d8657f7d#diff-d1a5ece2215eb348ea751cd0ac48592fR91)

The test seems to fail to create a pod with the volume and other tests that don't specify these mount options don't fail. So, I guess that this is caused because this environment doesn't support either of the option above.

Is it possible to try testing it manually by changing these options?

**Comment by runcom** [comment]
> Is it possible to try testing it manually by changing these options?

 yes, I have a box where I can change them and re-test, do you have suggestions on what to use to test?

**Comment by runcom** [comment]
(sorry fat fingers)

**Comment by runcom** [comment]
> If it was an unsupported mount option, then the test should fail every time, not just flake?

yes, it's failing every time for us

**Comment by msau42** [comment]
Are there kubelet node logs we can look at?

**Comment by runcom** [comment]
yes, we have artifacts for each job at https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-federated-results/pr-logs/pull/kubernetes-sigs_cri-o/1870/test_pull_request_crio_e2e_fedora/1368/

**Comment by msau42** [comment]
@mkimuram do you think we could make the test easier to debug and figure out which mount option doesn't work?

**Comment by runcom** [comment]
yes, I'm trying to add something to that command to at least have some logs if that fails cause right now, if you do kubectl get logs on the failing pod you don't get anything

**Comment by runcom** [comment]
@mkimuram looks like removing `nosuid` makes the container happy and test as well. I'm not sure why it fails on rhel and fedora though, I'll dig into that, but yeah, we could use something different if possible

**Comment by runcom** [comment]
confirmed that if I remove nosuid the test works fine, I"m modifying the test code to output something useful in case it fails

**Comment by mkimuram** [comment]
@runcom 

Thank you for your quick investigation and improvement!

**Comment by runcom** [comment]
I'm going to open a PR for the test output to have something useful (`|| mount | grep 'on /mnt/test'`) and understand why it fails with `nosuid`

**Comment by wongma7** [comment]
sorry, I am the one who wrote that hideous bash one-liner. TY for any patches

**Comment by runcom** [comment]
so, `nosuid` isn't shown:
```
[ec2-user@ip-172-18-7-20 kubernetes]$ cluster/kubectl.sh logs pvc-volume-tester-nggv8 -n e2e-tests-volumes-5r7dn
10.88.1.112:/export/pvc-9ddb9b00-d86e-11e8-af9a-0ef4de25ffb6 on /mnt/test type nfs4 (rw,nodev,relatime,vers=4.1,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,port=0,timeo=600,retrans=2,sec=sys,clientaddr=10.88.0.1,local_lock=none,addr=10.88.1.112)
```
is that required for the test? can we have another option checked instead? @mkimuram

**Comment by runcom** [comment]
maybe we can use `relatime`? who is setting `nosuid` and where?

**Comment by runcom** [comment]
PTAL at #70243 @msau42 @mkimuram @wongma7

**Comment by runcom** [comment]
> maybe we can use `relatime`? who is setting `nosuid` and where?

do you guys have any insight on this? @msau42 @mkimuram @wongma7

**Comment by mkimuram** [comment]
@runcom 

> maybe we can use relatime? who is setting nosuid and where?

As for this e2e test, we will be able to use any options which are valid for nfs e2e test, because the purpose of the test is to check that the volume mount options work well for provisioning and the options are properly reflected to the mount options inside the pod.

(Actually, I chose ```nosuid``` for nfs test just because I found it first.)

As for ```nosuid``` issue for nfs dynamic provisioner with this environment, it might be still worth looking into, if this option can be specified by users. However, this could be out of the scope of this issue.

**Comment by runcom** [comment]
> As for `nosuid` issue for nfs dynamic provisioner with this environment, it might be still worth looking into, if this option can be specified by users. However, this could be out of the scope of this issue.

exactly, thanks, I'm going to investigate it _not_ as part of this PR

**Comment by jberkus** [comment]
Since this test failure is non-blocking for release, setting priority approprately.

/priority important-longterm
